src/java/boa/aggregators/MeanAggregator.java

```diff
@@ -28,9 +28,17 @@
 
 	public void count(final String metadata) {
 		if (metadata == null)
+		{
 			this.count++;
+		}			
 		else
-			this.count += Long.parseLong(metadata);
+		{
+			try {
+				this.count += Long.parseLong(metadata);
+			} catch (NumberFormatException e) {
+				this.count += (long)Double.parseDouble(metadata);
```

Comment:
Here it calculates only the cardinality not the sum. The the sum is calculated inside the FloatMeanAggregator class inside. So the Method count inside MeanAggregator in Aggregator will calculate 1.0 + 1.0 + 1.0 = 3 and the sum is calculated inside the FloatMeanAggregator as following
```java

/** {@inheritDoc} */
	@Override
	public void aggregate(final double data, final String metadata) {
		this.sum += data;

		super.count(metadata);
	}
```
The metadata i found is either "1" or "1.0"